esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -2089,53 +2031,71 @@ void WsWuInfo::getWorkunitThorSlaveLog(const char *groupName, const char *ipAddr
     if (!rIO)
         throw MakeStringException(ECLWATCH_CANNOT_READ_FILE,"Cannot read file %s.",logName.str());
 
-    OwnedIFileIOStream ios = createBufferedIOStream(rIO);
+    Owned<IBufferedFileIOStream> ios = createBufferedFileIOStream(rIO);
     if (slaveNum > 0)
     {
-        StringBuffer line;
-        bool eof = false;
-        bool include = false;
+        readWorkunitLog(ios.get(), buf, outFile);
+    }
+    else
+    {//legacy wuid
+        appendIOStreamContent(buf, ios.get(), forDownload);
+    }
+}
+
+void WsWuInfo::readWorkunitLog(IBufferedFileIOStream* ios, MemoryBuffer& buf, const char* outFile)
+{
+    VStringBuffer startwuid("Started wuid=%s", wuid.str());
+    VStringBuffer endwuid("Finished wuid=%s", wuid.str());
 
-        VStringBuffer startwuid("Started wuid=%s", wuid.str());
-        VStringBuffer endwuid("Finished wuid=%s", wuid.str());
+    bool eof = false;
+    bool outputThisLine = false;
+    unsigned processID = 0;
 
-        const char *sw = startwuid.str();
-        const char *ew = endwuid.str();
+    StringBuffer line;
+    CWsWuFileHelper helper(nullptr);
+    Owned<IFileIOStream> outIOS = helper.createIOStreamWithFileName(outFile, IFOcreate);
+    while (!eof)
+    {
+        line.clear();
+        eof = ios->readLine(true, line);
+        if (line.isEmpty())
+            continue;
 
-        Owned<IFileIOStream> outIOS;
-        if (!isEmptyString(outFile))
+        if (strstr(line, startwuid))
         {
-            CWsWuFileHelper helper(nullptr);
-            outIOS.setown(helper.createIOStreamWithFileName(outFile, IFOcreate));
+            outputThisLine = true;
+            outputALine(line.length(), line.str(), buf, outIOS);
```

Comment:
see question about 'buf' above. Afaics, readWorkunitLog is only ever called with a stream, buf is not used.

If so, it would clearer to call:
```
        outIOS->write(line.length(), line.str());
```
directly here